### ISSUE TITLE
Move: grabbed view is on top of layer-shell surfaces

### ISSUE THREAD

**Issue by dkondor** [WayfireWM/wayfire#1044]
**Describe the bug**
When moving a view by dragging it, it is displayed on top of layer-shell surfaces on the top and overlay layers.

**To Reproduce**
1. Start a layer-shell client that is on the top or overlay layers (I used the gtk-layer-demo, but the same happens with wf-panel as well)
2. Move a view by dragging it (either from the title bar or with the move binding)
3. The dragged view is on top, above the layer-shell surfaces

**Expected behavior**
The dragged view is still behind layer-shell surfaces that are normally above it (top and overlay layers). Works this way on aa3e078d830d93357a9c1e435e47e5633cf7f9a2

**Screenshots or stacktrace**

![layer-shell-drag](https://user-images.githubusercontent.com/6224679/104830995-5b60cb00-58bf-11eb-80cd-a4319832b3f9.png)

**Wayfire version**
git master

**Comment by damianatorrpm** [comment]
glad it's not only me :+1:

**Comment by ammen99** [comment]
Yeah, grabbed views are drawn basically as drag-and-drop icons on top of everything else to support cases like:

![slurped](https://user-images.githubusercontent.com/6287487/104835119-a0c3cf00-58a4-11eb-8313-6c9e20bbff00.png)

Notice that the view there is visible on top of all workspaces, and even in places where there is no workspace.

Fixing this will require some special-casing - we'll need to support both the old way of dragging views (drawing them with mirror views) and the new one (with an overlay). The move-drag-interface will also need a new signal to communicate with plugins to determine which variant to use for each output.

Would it make it clearer what is going on if the view were made semi-transparent? (to indicate it is being dragged and is no longer in "normal mode"?

**Comment by damianatorrpm** [comment]
I understand the suggestion, since as @ammen99 said on irc, the behaviour described in this issue,
massively simplifies implementation details.

Personally I find it strange that moving views go above layer-shells, maybe because of habit and usage
of other desktop environments (afaik none do this), but maybe I'll get used to this behaviour so 
I don't want to be to noisy about one way or the other.

**Comment by dkondor** [comment]
I see, this makes a lot of sense -- this does not seem like a very important issue, although I have a slight preference for the original behavior :)
Maybe change this to "enhancement" and revisit it later?

**Comment by ammen99** [comment]
> slight preference for the original behavior :)

Understandable. I'll leave this for later, hoping that I get a smart idea for doing less work in the meantime :)

**Comment by travankor** [comment]
I kind of (slightly) prefer the new behavior l. Compared to other wlroots compositors, it is kind of strange but you get used to it. Another advantage is that moving a smaller view over a larger layer shell view doesn’t get hidden.

Maybe instead of changing behavior, just add an effect like fade when moving over views that are supposed to be on top?
